### DISCUSSION TITLE
Socket.io+supabase = single API?

### DISCUSSION THREAD

**Discussion by coderextreme** [socketio/socket.io#5183] [Ideas]
Can socket.io and supabase be put into a single API, like a supabase transport for socket.io?  I know socket.io has more language support.  I’m interested in JavaScript.  Specifically, web browser clients.

**Comment by darrachequesne** [comment]
That's a great question! I know they are using plain WebSocket for their realtime module: https://github.com/supabase/realtime-js/

Not sure if they expose a way to provide a custom low-level transport. Which feature would you need from supabase?

**Comment by coderextreme** [comment]
Just a way to use my existing socket.io software without a total rewrite.

However, I am thinking adding room tokens to gain access to a room. So two
people can have two rooms with the same name, ala Discord.

Perhaps an Elixir server-side port is due?

There’s a lot more, including storage of metaverse worlds.  I can probably
use a redis adapter for that.

We’ve only got a base MVP at this point,
https://github.com/coderextreme/JSONverse

**Reply by coderextreme** [reply]
I think the most useful thing to add from supabase would be PostgreSQL support!
